### PR TITLE
Adding vscode devcontainer support

### PR DESCRIPTION
**Pull request by lumpi1982**
To easily contribute devcontainer support is nice. I took the code which is originally provided by https://github.com/custom-components/integration_blueprint

### CODE REVIEW

**Comment by DarwinsBuddy** [comment]
Thanks a lot for your contribution.

I am currently on vacation and won't be able to look at it for another 2 weeks. But rest assured, I'll come back to it asap. :)

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```
Comment (DarwinsBuddy) [inline .vscode/launch.json:2]: json (without extension) does not support comments. please remove all comments from this file :)

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "version": "0.2.0",
+    "configurations": [
+      {
+        // Example of attaching to local debug server
```
Comment (DarwinsBuddy) [inline .vscode/launch.json:6]: see above

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        "host": "localhost",
+        "pathMappings": [
+          {
+            "localRoot": "${workspaceFolder}",
+            "remoteRoot": "."
+          }
+        ]
+      },
+      {
+        // Example of attaching to my production server
```
Comment (DarwinsBuddy) [inline .vscode/launch.json:20]: see above

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
+// See https://aka.ms/vscode-remote/devcontainer.json for format details.
```
Comment (DarwinsBuddy) [inline .devcontainer/devcontainer.json:1]: json (without extension) does not support comments. please remove all comments from this file :)

**Review by DarwinsBuddy — CHANGES_REQUESTED** [review]
I tried to follow the instructions, but unfortunately I did not get asked to "reopen the project". Is there another way to the build? Can you document it?

**Comment by lumpi1982** [comment]
> I tried to follow the instructions, but unfortunately I did not get asked to "reopen the project". Is there another way to the build? Can you document it?

Do you get any specific error message? Does a sample container i.e. https://github.com/microsoft/vscode-remote-try-python work for you?

**Comment by DarwinsBuddy** [comment]
Thanks for the commit
I got the popup now, but unfortunately the devcontainer build now hangs on 
```
[2022-12-04T13:39:55.269Z] Error: Command failed: docker run --sig-proxy=false -a STDOUT -a STDERR -p 9123:8123 --mount type=bind,source=/home/spoerk/dev/sandbox/lumpi1982/WienerNetzeSmartmeter,target=/workspaces/WienerNetzeSmartmeter --mount type=volume,src=vscode,dst=/vscode -l devcontainer.local_folder=/home/spoerk/dev/sandbox/lumpi1982/WienerNetzeSmartmeter --entrypoint /bin/sh vsc-wienernetzesmartmeter-55b9299f186fa2c790430b7df2bac9b7-features -c echo Container started
[2022-12-04T13:39:55.269Z] trap "exit 0" 15
[2022-12-04T13:39:55.269Z] exec "$@"
[2022-12-04T13:39:55.269Z] while sleep 1 & wait $!; do :; done -
[2022-12-04T13:39:55.270Z]     at loe (/home/spoerk/.vscode/extensions/ms-vscode-remote.remote-containers-0.262.3/dist/spec-node/devContainersSpecCLI.js:1887:1339)
[2022-12-04T13:39:55.270Z]     at eT (/home/spoerk/.vscode/extensions/ms-vscode-remote.remote-containers-0.262.3/dist/spec-node/devContainersSpecCLI.js:1887:1275)
[2022-12-04T13:39:55.270Z]     at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
[2022-12-04T13:39:55.270Z]     at async voe (/home/spoerk/.vscode/extensions/ms-vscode-remote.remote-containers-0.262.3/dist/spec-node/devContainersSpecCLI.js:1893:2049)
[2022-12-04T13:39:55.270Z]     at async Xf (/home/spoerk/.vscode/extensions/ms-vscode-remote.remote-containers-0.262.3/dist/spec-node/devContainersSpecCLI.js:1893:3212)
[2022-12-04T13:39:55.271Z]     at async Jae (/home/spoerk/.vscode/extensions/ms-vscode-remote.remote-containers-0.262.3/dist/spec-node/devContainersSpecCLI.js:2013:15058)
[2022-12-04T13:39:55.271Z]     at async Wae (/home/spoerk/.vscode/extensions/ms-vscode-remote.remote-containers-0.262.3/dist/spec-node/devContainersSpecCLI.js:2013:14812)
[2022-12-04T13:39:55.276Z] Stop (56206 ms): Run: /opt/visual-studio-code/code --ms-enable-electron-run-as-node /home/spoerk/.vscode/extensions/ms-vscode-remote.remote-containers-0.262.3/dist/spec-node/devContainersSpecCLI.js up --user-data-folder /home/spoerk/.config/Code/User/globalStorage/ms-vscode-remote.remote-containers/data --workspace-folder /home/spoerk/dev/sandbox/lumpi1982/WienerNetzeSmartmeter --workspace-mount-consistency cached --id-label devcontainer.local_folder=/home/spoerk/dev/sandbox/lumpi1982/WienerNetzeSmartmeter --log-level debug --log-format json --config /home/spoerk/dev/sandbox/lumpi1982/WienerNetzeSmartmeter/.devcontainer/devcontainer.json --default-user-env-probe loginInteractiveShell --mount type=volume,source=vscode,target=/vscode,external=true --skip-post-create --update-remote-user-uid-default on --mount-workspace-git-root true
[2022-12-04T13:39:55.277Z] Exit code 1
[2022-12-04T13:39:55.283Z] Command failed: /opt/visual-studio-code/code --ms-enable-electron-run-as-node /home/spoerk/.vscode/extensions/ms-vscode-remote.remote-containers-0.262.3/dist/spec-node/devContainersSpecCLI.js up --user-data-folder /home/spoerk/.config/Code/User/globalStorage/ms-vscode-remote.remote-containers/data --workspace-folder /home/spoerk/dev/sandbox/lumpi1982/WienerNetzeSmartmeter --workspace-mount-consistency cached --id-label devcontainer.local_folder=/home/spoerk/dev/sandbox/lumpi1982/WienerNetzeSmartmeter --log-level debug --log-format json --config /home/spoerk/dev/sandbox/lumpi1982/WienerNetzeSmartmeter/.devcontainer/devcontainer.json --default-user-env-probe loginInteractiveShell --mount type=volume,source=vscode,target=/vscode,external=true --skip-post-create --update-remote-user-uid-default on --mount-workspace-git-root true
[2022-12-04T13:39:55.283Z] Exit code 1
```

**Comment by DarwinsBuddy** [comment]
you may want to rebase the PR :) pulled the dependency into main to separate it from this feature

**Comment by DarwinsBuddy** [comment]
I have to close this due to inactivity. @TheRealVira introduced this topic.

Still, thanks for the contribution. :)

**Pull request closed without merging**